### PR TITLE
Update GitHub Actions

### PR DESCRIPTION
**Pull request by NikolaiMadlener**
# *Update Github Actions*

## :gear: Release Notes 
- update GitHub actions (remove watchOS, visionOS and iPadOS packages)

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [ ] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -20,63 +20,28 @@ jobs:
...
-    name: Build and Test Swift Package visionOS
-    uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
-    with:
-      xcodeversion: latest
-      scheme: TemplatePackage
-      resultBundle: TemplatePackageVisionOS.xcresult
-      destination: 'platform=visionOS Simulator,name=Apple Vision Pro'
-      artifactname: TemplatePackageVisionOS.xcresult
+      scheme: SpeziLicense
+      artifactname: SpeziLicense.xcresult
```
Comment (PSchmiedmayer) [inline .github/workflows/build-and-test.yml:24]: As we have multiple targets in the Swift Package you will need to use:
```suggestion
      scheme: SpeziLicense-Package
      artifactname: SpeziLicense-Package.xcresult
```

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -20,63 +20,28 @@ jobs:
...
     name: CodeQL
     uses: StanfordBDHG/.github/.github/workflows/xcodebuild-or-fastlane.yml@v2
     with:
       codeql: true
       xcodeversion: '14.3.1'
-      scheme: TemplatePackage
+      scheme: SpeziLicense
     permissions:
       security-events: write
       actions: read
```
Comment (PSchmiedmayer) [inline .github/workflows/build-and-test.yml:41]: CodeQL is currently not working for Swift 5.10 as far as I am aware of so should probably remove this job.

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
Thank you for looking into this @NikolaiMadlener!

I would suggest to merge this with the #1 PR and do all the changes in a single PR 👍

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (NikolaiMadlener) [reply .github/workflows/build-and-test.yml:24]: So this is also if we just have one target and one test-target? I just checked other repos like SpeziOnboarding, where it's the same case. There, the "-Package" suffix is not used, that's why I did it the same way.
But can change it of course.

Comment (PSchmiedmayer) [reply .github/workflows/build-and-test.yml:24]: You are completely right; I missed why this error is appearing here. I thought it was due to having multiple targets in the Swift Package (which is when the `-Package`) postfix needs to be added as Xcode generates one package scheme for all targets.

This error seems to stem from the fact that you have not renamed the package target but expected it to be called `SpeziLicense` while it is still called `TemplatePackage ` in the Package.swift file.

As we will close this PR, you can take these GitHub Actions and put them in the other branch and everything should work as expected as you have renamed the target there 👍

**Comment by PSchmiedmayer** [comment]
Closed in favor of #1.

**Pull request closed without merging**